src/main/java/io/jmix/uisamples/view/flowui/components/virtuallist/inlineeditor/VirtualListInlineEditor.java

```diff
@@ -0,0 +1,195 @@
+                    .build()
+                    .open());
+            detailButton.addThemeVariants(ButtonVariant.LUMO_TERTIARY_INLINE);
+
+            Button removeButton = new Button(new Icon(VaadinIcon.TRASH));
+            removeButton.setText(messages.getMessage("actions.Remove"));
+            removeButton.addThemeVariants(ButtonVariant.LUMO_TERTIARY_INLINE, ButtonVariant.LUMO_ERROR);
+            removeButton.addClickListener(e -> {
+                foodDc.getMutableItems().remove(item);
+                dataContext.remove(dataContext.merge(item));
+            });
+
+
+            buttonsPanel.add(detailButton, removeButton);
+            rootCardLayout.add(avatar, infoLayout, buttonsPanel);
+            return rootCardLayout;
+        });
+
+    }
+
+    @Install(to = "foodDl", target = Target.DATA_LOADER)
+    private List<Food> foodDlLoadDelegate(final LoadContext<Food> loadContext) {
+        return getInitialFoodList();
+    }
+
+    private Food createFoodSample(String title,
+                                  String description,
+                                  int price,
+                                  String iconLocation) {
+
+        Food food = dataContext.create(Food.class);
+        food.setTitle(title);
+        food.setDescription(description);
+        food.setPrice(price);
+        try (InputStream imageStream = resources.getResourceAsStream(iconLocation)) {
+            food.setIcon(imageStream.readAllBytes());
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot open a stream", e);
+        }
+        return food;
+    }
+
+    private List<Food> getInitialFoodList() {
+        List<Food> foodList = new ArrayList<Food>();
+        foodList.add(createFoodSample(messages.getMessage(getClass(), "sample.burger.title"),
+                messages.getMessage(getClass(), "sample.burger.description"),
+                7, "META-INF/resources/icons/components/virtuallist-food/burger.png"));
+
+        foodList.add(createFoodSample(messages.getMessage(getClass(), "sample.cola.title"),
+                messages.getMessage(getClass(), "sample.cola.description"),
+                4, "META-INF/resources/icons/components/virtuallist-food/cola.png"));
+
+        foodList.add(createFoodSample(messages.getMessage(getClass(), "sample.fries.title"),
+                messages.getMessage(getClass(), "sample.fries.description"),
+                3, "META-INF/resources/icons/components/virtuallist-food/fries.png"));
+
+        foodList.add(createFoodSample(messages.getMessage(getClass(), "sample.pizza.title"),
+                messages.getMessage(getClass(), "sample.pizza.description"),
+                15, "META-INF/resources/icons/components/virtuallist-food/pizza.png"));
+
+        foodList.add(createFoodSample(messages.getMessage(getClass(), "sample.sandwich.title"),
+                NOOP_DESCRIPTION,
+                2, "META-INF/resources/icons/components/virtuallist-food/sandwich.png"));
```

Comment:
`META-INF/resources/icons/components` has already been used several times, maybe move it to a separate static variable? `ICONS_PATH_ROOT`?